hls4ml/writer/vivado_writer.py

```diff
@@ -230,6 +230,41 @@ def write_weights(self, model):
         for layer in model.get_layers():
             for weights in layer.get_weights():
                 self.print_array_to_cpp(weights, model.config.get_output_dir())
+    
+    def __make_dat_file(self, model, filepath, is_input): 
```

Comment:
I would prefer if `is_input` is replaced in favor of the filename to which data will be written.